src/main/java/duke/commands/FindCommand.java

```diff
@@ -0,0 +1,43 @@
+package duke.commands;
+
+import duke.data.TaskList;
+
+import java.lang.StringBuilder;
+
+/**
+ * Implements the find command.
+ * @author Lim Yong Shen, Kevin
+ */
+public class FindCommand extends Command {
+
+    private String keyword;
+    public static final String COMMAND_WORD = "find";
+
+    /**
+     * Constructs a find command with the specified keyword.
+     * @param keyword The specified keyword.
+     */
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    /**
+     * Executes this FindCommand and returns its CommandResult.
+     * @return This FindCommand's CommandResult.
+     */
+    @Override
+    public CommandResult execute() {
```

Comment:
Maybe it is better to put the find functionality with the substring method in FindCommand rather than TaskList?

Reply:
That's true... I'll keep this in mind when refactoring the code. I'll look into cleaning up the code after implementing the GUI :)